src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -152,16 +152,16 @@ typedef cpuset_t cpu_set_t;
     PER_FUNCTION_BLOCK(numa_node_of_cpu)
 
 // Declare pointers to all the used numa functions
-#define PER_FUNCTION_BLOCK(fn) extern decltype(fn)* fn##_ptr;
+#define PER_FUNCTION_BLOCK(fn) extern decltype(fn)* fn##_ptr_gc;
```

Comment:
hoping these `NUMA` collisions between gc_pal and coreclr_pal can be avoided in a better way.

Reply:
Can we extract out numshim in a separate static object (and link it into coreclr and nativeaot main binaries)? That will provide symbols for both, GC and PAL without duplication.

Reply:
yeah that should be possible. @janvorli ?

Reply:
That sounds like a good idea. And I would do the same for the cgroup stuff too. It would be great to finally not to have the cgroup stuff duplicated. In fact, there is a third clone of the cgroup in nativeaot, so it would be even nicer to have it in one location.

Reply:
Is the Numa support needed by anything outside the GC? I think the Numa support in the Win32 PAL can be deleted and we can have it in the GC PAL only.

Reply:
It is used by NumaNodeInfo in utilcode. While it is currently used for the built-in GC only, we have plans to work on making threadpool NUMA aware too so I expect this to be used in future.

Reply:
I would make the GC to be the source of truth for memory partitioning for this, and provide that information to the rest of the system. I do not think it makes sense to duplicate it. It would just lead to mismatches.

Reply:
I don't think it was suggested to duplicate it, but rather de-duplicate it by putting it into a separate static library so it would be neither part of GC nor the coreclr / native aot app. My understanding was that the standalone GC, the coreclr and the native aot app would then link to this library.

Reply:
I am suggesting to make the NUMA PAL part of the GC PAL so that we only need to worry how the GC talks about it. We do not need to worry about factoring it into a separate library with this plan.

If some other part of the systems wants to know the details about memory partitioning, we would introduce GC API to get it. The answer would include all potential policies that GC applied on top of it (like config overrides, adjustments for unbalanced configs, etc.).

Reply:
agree that we should do better factoring of the NUMA stuff, but hoping to keep this specific PR minimal so we could possibly backport to 7. We already hit some compat issues with separating the cgroup stuff. Does that sound reasonable?

Reply:
If you are looking for a safe .NET 7 change, you should just do very targeted fix for the path that is causing the perf problem (ie create more duplicated code). I do not think that backporting the GC PAL switch to servicing is a good idea.

Reply:
have removed `numa.cpp` from the PAL. Guessing more things could be refactored subsequently particularly the cgroup stuff but that can happen as a separate change.